R/add_sda_target.R

```diff
@@ -0,0 +1,185 @@
+#' Add portfolio level SDA target
+#'
+#' This function calculates the portfolio-level emission factor targets, as
+#' calculated using the sectoral-decarbonisation approach.
+#'
+#' @param data A dataframe like the output of
+#'   [r2dii.match::prioritize()].
+#' @param ald An asset level dataframe like [r2dii.data::ald_demo].
+#' @param co2_intensity_scenario A scenario dataframe like
+#'   [r2dii.analysis::co2_intensity_scenario].
+#'
+#' @return  A tibble with portfolio-level emission_factor targets.
+#' @export
+#'
+#' @examples
+#' installed <- requireNamespace("r2dii.data", quietly = TRUE) &&
+#'   requireNamespace("r2dii.match", quietly = TRUE)
+#' if (!installed) stop("Please install r2dii.match and r2dii.data")
+#'
+#' library(r2dii.data)
+#' library(r2dii.match)
+#'
+#' valid_matches <- match_name(loanbook_demo, ald_demo) %>%
+#'   # WARNING: Remember to validate matches (see `?prioritize`)
+#'   prioritize()
+#'
+#' valid_matches %>%
+#'   add_sda_target(
+#'     ald = ald_demo,
+#'     co2_intensity_scenario = r2dii.analysis::co2_intensity_scenario
+#'   )
+add_sda_target <- function(data, ald, co2_intensity_scenario) {
+  start_year <- co2_intensity_scenario %>%
+    select(.data$sector, .data$year) %>%
+    group_by(.data$sector) %>%
+    summarize(start_year = min(.data$year))
+
+  ald <- ald %>%
+    left_join(start_year, by = "sector") %>%
+    filter(.data$year >= .data$start_year) %>%
+    filter(!is.na(.data$emission_factor), !is.na(.data$production)) %>%
+    select(-.data$start_year)
+
+
+  ald_sda_market_benchmark <- calculate_sda_market_benchmark(
```

Comment:
Maybe remove the prefix `calculate_`. The pro of making it a verb is outweighed by the con of a function name that is probably too long. Long names are generally not a problem but they may obscure the intent of the code.